digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/IntegrationApi.scala

```diff
@@ -240,6 +246,33 @@ class IntegrationApi(val massTransitStopService: MassTransitStopService) extends
         .getOrElse("")
   }
 
+  def lastModifiedBy(createdBy: Option[String], modifiedBy: Option[String]): (String, Boolean) = {
+
+    val nonFixedValues = List("split_speedlimit_.*", "batch_process_.*")
+
+    val nonFixedModified = nonFixedValues match {
+      case value if value.exists(nfv => nfv.r.findFirstIn(modifiedBy.getOrElse("")).equals(modifiedBy)) =>
+        modifiedBy
+      case _ => None
+    }
+    val nonFixedCreated = nonFixedValues match {
+      case value if value.exists(nfv => nfv.r.findFirstIn(modifiedBy.getOrElse("")).equals(modifiedBy)) =>
+        createdBy
+      case _ => None
+    }
+
+    val autoGeneratedValues = List("dr1conversion", "dr1_conversion", nonFixedModified.getOrElse(""), nonFixedCreated.getOrElse(""), "automatic_correction", "excel_data_migration", "automatic_generation", "vvh_generated")
+
+    modifiedBy match{
+      case None => "generatedValue" -> false
+      case Some(value) if(value != null) => if (autoGeneratedValues.exists(agv => agv.equals(modifiedBy.getOrElse(""))))return "generatedValue" -> true else return "generatedValue" -> false
+    }
```

Comment:
Can you try like this: 

```
modifiedBy match{

   case Some(value) if value != null && autoGeneratedValues.exists(agv => agv.equals(modifiedBy.getOrElse(""))) => 

     "generatedValue" -> true 

   case None => "generatedValue" -> false
```

I believe that with this you can remove the `else return "generatedValue" -> false`